templates/katello.yaml.erb

```diff
@@ -31,6 +31,10 @@
     :url: <%= @qpid_url %>
     :subscriptions_queue_address: <%= @candlepin_event_queue %>
 
+  :container_image_registry:
+    :crane_url: <%= @crane_url %>
```

Comment:
Why are these prefixed with `crane_`? I'd expect them to be `url` and `ca_cert_file` (similar to others above).

Reply:
These are the configs that katello uses to find the crane registry. These are not configs that are exposed to the users of the registry.

Reply:
Well, they are exposed through this module.

Also it doesn't answer why this is pinned to crane. If this option only supports crane and not a generic registry then I'd expect that the section is named `crane`. The alternative is that it's generic and any registry is supported. In both cases it doesn't make sense to me to prefix options with `crane_`.

Reply:
This section is to config the aspects of the newly implemented registry endpoint. As part of the implementation, katello needs to call pulp's crane; that's what these two configs are for. Removing the _crane_ prefix would then not convey what the url is meant to point at.

If preferred, I could add another entry to indicate the type of backend registry it is pointing at and then remove the _crane_ prefix from the existing two entries. Please suggest a name for this new entry.

Reply:
It would help me to see how this section and its values are used. The PR mentioned in the opening message is merged but since then I think it has changed. Could you describe how it's used or link to the current code?

Reply:
Usage:
Don't expose registry if settings absent
https://github.com/Katello/katello/blob/master/app/controllers/katello/api/registry/registry_proxies_controller.rb#L410-L415

Pull objects from crane
https://github.com/Katello/katello/blob/master/app/lib/katello/resources/registry.rb#L22-L27